### ISSUE TITLE
Potential bug in image_processing.py

### ISSUE THREAD

**Issue by anchwr** [dnarayanan/powderday#105]
I'm getting an error (TypeError: No conversion path for dtype: dtype('<U13')) when I run image_processing.py. The issue seems to be in the convolve function, where the names of the filters that have been used are saved to an HDF5 file along with the relevant data. It looks like there was an update earlier this year that used utf-8 encoding with the filter names so that they could be processed by HDF5. However, the list of filter names being saved to the HDF5 file ('trimmed_names') is not the one that contains the encoded names ('names'). Using the list of encoded names seems to solve the issue. I've linked the relevant lines below:
https://github.com/dnarayanan/powderday/blob/e4188330dec06d0d7fe51924d996658a794e788c/powderday/image_processing.py#L94-L97

**Comment by dnarayanan** [comment]
ah! i think this may be the same error uncovered by jeremy bailin (via email, though i was hoping that they may issue a PR).  can you try the following and see if it works for you?  if so please update and i'll issue a new commit (or you can issue a PR either way)

```

Also, we discovered what might be a bug in the image generation: we got an h5py error that seems to be related to these lines in image_processing.py:

# Don't add the names of filters that were skipped
trimmed_names = list(set(filterfilenames) - set(skip_conv))
# Encode in utf8 so hdf5 file can process and append names (doesn't accept strings)
names = [n.encode('utf8') for n in trimmed_names]
f.create_dataset("filter_names", data=trimmed_names)

It seems like names gets defined with the appropriate encoding, but then never gets used, and we're getting a unicode-looking error. We're replacing the last line with:

f.create_dataset("filter_names", data=names)

and seeing if that works. :)

Cheers,
-Jeremy.```

**Comment by anchwr** [comment]
Ah, yep! That's the same change I made :) It works. I just made a PR request.

**Comment by dnarayanan** [comment]
fixed in commit e2a43d3f6b3b2e038e23ece97093a1247f0059ba via the PR from @anchwr
